cmd/githubwebhookserver/main.go

```diff
@@ -99,9 +100,18 @@ func main() {
 	flag.StringVar(&c.BasicauthUsername, "github-basicauth-username", c.BasicauthUsername, "Username for GitHub basic auth to use instead of PAT or GitHub APP in case it's running behind a proxy API")
 	flag.StringVar(&c.BasicauthPassword, "github-basicauth-password", c.BasicauthPassword, "Password for GitHub basic auth to use instead of PAT or GitHub APP in case it's running behind a proxy API")
 	flag.StringVar(&c.RunnerGitHubURL, "runner-github-url", c.RunnerGitHubURL, "GitHub URL to be used by runners during registration")
+	flag.StringVar(&logFormat, "log-format", "text", `The log format. Valid options are "text" and "json". Defaults to "text"`)
+	
 
 	flag.Parse()
 
+	setupLog, err := logging.NewLogger(logLevel, logFormat)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: creating logger: %v\n", err)
+		os.Exit(1)
+	}
+	setupLog.WithName("setup")
```

Comment:
The idea behind naming it `setupLog` was to use the logger for any error reporting up until you finally parse the log level to generate the non-setup logger. As you're going to change the design to use `fmt.Fprintf` instead of the former `setupLog`, I think we can safely rename this from `setupLog` to `log` now 🤔

Reply:
updated